### PR TITLE
MAINT: move coverage config to setup.cfg

### PR DESCRIPTION
**Pull request by kohr-h**
Little addition to #1472: Move the `coverage` configuration to `setup.cfg`, which helps us get rid of one top-level file. This is officially supported according to [the documentation](https://coverage.readthedocs.io/en/latest/config.html?highlight=setup.cfg).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -111,3 +112,55 @@ max-line-length = 79
...
     __pycache__
+
+[coverage:run]
+branch = False
+include =
+    odl/*
+omit =
+    */__init__.py
+    odl/diagnostics/*
+    odl/util/*
```
Comment (adler-j) [inline setup.cfg]: Why do we omit this?

Comment (kohr-h) [reply setup.cfg]: I don't know. My guess is that it's from a time when the utility code was really only for internal usage and rather small in scope. Mostly little helpers and testing stuff. I'll check what makes sense to exclude there.

Comment (kohr-h) [reply setup.cfg]: Now more fine-grained. And `tests/` excluded, they used to drag down our coverage score.

**Comment by kohr-h** [comment]
Some more last changes. What I can't get rid of is the `def` and similar lines showing up as not covered. According to [the coverage FAQ](https://coverage.readthedocs.io/en/v4.5.x/faq.html) this shouldn't happen if called from the command line.

But anyway, the reported coverage is probably closer to reality now :-)
I'll merge after the final CI run has finished.

**Comment by kohr-h** [comment]
I got one last idea that seems to work locally. Let's see how we'll do.

**Comment by kohr-h** [comment]
Okay, I've solved the problem. If I run `coverage` manually (not through `pytest-cov`) the `def` and `class` lines etc. are correctly reported as covered. So I've changed `.travis.yml` to spit out a temporary script to run the tests (basically `import odl; odl.test()`) and to run it with `coverage run`.

**Pull request merged**